Add flags currency

gemdev111 opened the pull request:
added flags for supported currencies natively for iOS, updated settings selected currency with flag

gemdev111 committed:
Add flags to each currency #14 https://github.com/gemwalletcom/gem-ios/issues/14

gemdev111 committed:
Add flags to each currency  #14 https://github.com/gemwalletcom/gem-ios/issues/14

gemcoder21 commented on Gem/Settings/ViewModels/CurrencySceneViewModel.swift:
```diff
@@ -66,6 +70,70 @@ class CurrencySceneViewModel: ObservableObject {
 extension Locale.Currency {
     var title: String {
-        return String(format: "%@ - %@", identifier, Locale.current.localizedString(forCurrencyCode: identifier) ?? .empty)
+        let localizedCurrencyId = Locale.current.localizedString(forCurrencyCode: identifier) ?? .empty
```
can you add unit test for this one? to make sure we properly present the title

gemdev111 replied:
will add

gemcoder21 commented on Gem/Settings/ViewModels/CurrencySceneViewModel.swift:
```diff
@@ -66,6 +70,70 @@ class CurrencySceneViewModel: ObservableObject {
+    }
+
+    fileprivate static func flag(for identifier: String) -> String? {
+        return Self.emojiFlags[identifier] ?? ""
```
if you return "String?` should we avoid returning `?? ""`

gemdev111 replied:
correct, typo

gemcoder21 reviewed (approved):
would be great to attach screenshot to see how it looks

gemdev111 committed:
[-] removed unused optional unwrapping with empty string
[+] added "₿" for future implementation in currency core
[+] added CurrencySceneViewModelTests (test the title for currency)

gemdev111 committed:
Merge branch 'main'

Conflicts:
	Gem.xcodeproj/project.pbxproj

gemdev111 committed:
[f] xcode pbproj fixes

gemdev111 commented:
> would be great to attach screenshot to see how it looks
![image](https://github.com/gemwalletcom/gem-ios/assets/171273137/8591710a-3072-4b59-b72e-1558e65a32d8)
![image](https://github.com/gemwalletcom/gem-ios/assets/171273137/981ee18b-f6f4-4a2f-9d0a-58de929327a3)
